seqr/utils/search/hail_search_utils_tests.py

```diff
@@ -0,0 +1,191 @@
+        {'sample_id': 'HG00733', 'individual_guid': 'I000006_hg00733', 'family_guid': 'F000002_2', 'project_guid': 'R0001_1kg', 'affected': 'N', 'sex': 'F'}
+    ],
+}
+CUSTOM_AFFECTED_SAMPLE_DATA = {'VARIANTS': deepcopy(EXPECTED_SAMPLE_DATA['VARIANTS'])}
+CUSTOM_AFFECTED_SAMPLE_DATA['VARIANTS'][0]['affected'] = 'N'
+CUSTOM_AFFECTED_SAMPLE_DATA['VARIANTS'][1]['affected'] = 'A'
+CUSTOM_AFFECTED_SAMPLE_DATA['VARIANTS'][2]['affected'] = 'U'
+
+FAMILY_1_SAMPLE_DATA = {
+    'VARIANTS': [
+        {'sample_id': 'NA19675', 'individual_guid': 'I000001_na19675', 'family_guid': 'F000001_1', 'project_guid': 'R0001_1kg', 'affected': 'A', 'sex': 'M'},
+        {'sample_id': 'NA19678', 'individual_guid': 'I000002_na19678', 'family_guid': 'F000001_1', 'project_guid': 'R0001_1kg', 'affected': 'N', 'sex': 'M'},
+    ],
+}
+
+
+@mock.patch('seqr.utils.search.hail_search_utils.HAIL_BACKEND_SERVICE_HOSTNAME', MOCK_HOST)
+class HailSearchUtilsTests(SearchTestHelper, TestCase):
+    databases = '__all__'
+    fixtures = ['users', '1kg_project', 'reference_data']
+
+    def setUp(self):
+        super(HailSearchUtilsTests, self).set_up()
+
+    def _test_expected_search_call(self, search_fields=None, gene_ids=None, intervals=None, exclude_intervals= None,
+                                   rs_ids=None, variant_ids=None, dataset_type=None, secondary_dataset_type=None,
+                                   frequencies=None, custom_query=None, quality_filter=None, inheritance_mode='de_novo',
+                                   sort='xpos', sort_metadata=None, num_results=100, sample_data=None, omit_sample_type=None):
+        sample_data = sample_data or EXPECTED_SAMPLE_DATA
+        if omit_sample_type:
+            sample_data = {k: v for k, v in sample_data.items() if k != omit_sample_type}
+
+        expected_search = {
+            'requester_email': 'test_user@broadinstitute.org',
+            'sample_data': sample_data,
+            'genome_version': 'GRCh37',
+            'sort': sort,
+            'sort_metadata': sort_metadata,
+            'num_results': num_results,
+            'inheritance_mode': inheritance_mode,
+            'inheritance_filter': {},
+            'dataset_type': dataset_type,
+            'secondary_dataset_type': secondary_dataset_type,
+            'frequencies': frequencies,
+            'quality_filter': quality_filter,
+            'custom_query': custom_query,
+            'intervals': intervals,
+            'exclude_intervals': exclude_intervals,
+            'gene_ids': gene_ids,
+            'variant_ids': variant_ids,
+            'rs_ids': rs_ids,
+        }
+        expected_search.update({field: self.search_model.search[field] for field in search_fields or []})
+
+        request_body = json.loads(responses.calls[-1].request.body)
+        self.assertDictEqual(request_body, expected_search)
+
+    @responses.activate
+    def test_query_variants(self):
```

Comment:
`parsedLocus` is added in the intermediate search utils, its not something that ever part of the search model If you pass a `locus` it will be turned into `parsedLocus`  and then turned again into `intervals` in the request body. This is tested several times here